### PR TITLE
Fix main page

### PR DESCRIPTION
**Pull request by xbpcb**
Fix Main (Homepage) styles for both mobile and desktop.

Work in progress.

### CODE REVIEW

**Comment by MartinKolarik** [comment]
Not sure if this PR handles it, probably not - we're using custom grid size, and it's currently implemented by overriding the container widths all over the place. Instead, we should change the variables in `src/public/less/variables.less` the grid on the largest size is 12 * 80 + 11 * 24 in between, meaning 1224 px + padding around for the largest container.

File: src/public/img/about/about-gradient.svg [hunk 1]
```diff
@@ -0,0 +1,9 @@
...
...YUbRTZ7CiHbwAXwoys4N9brolaQxc+8Cz6MAGg7C5DmXvxASpVEWCfQJaVbmQAY0F5CJEDFzfzj4K9mufXuIPIpQwbC+avgYvb9/rbCnP7uqsGxFR1zyQXERXC76bPtc89QX6HmjFARAdg3Z4ZSO6ASWtTY44kwMmwNcQN7loofOiHTr1oiutO6zqkZZhUCouYSPY0vgdtikcgiBBd64OA2uXh2sQp7IgxJDkVWuxJJWkgQ/TsfAd43VmEEJ8Qs5vlkkwTNqmbZfHBQQvCNnXryBzK16ReRvuFgFAFE90Ewj+Dpevnjl1Hr73ci/FFqHPLJN/MB5XJlo+n2M0gu9fcFgmSMJh3SixZx7JyTAH8r7o3OXPKjxXMsfA9d8Tyej7jL4V+1A90FR6SDnsdOsQ0X/zdWM0Tctcop6Wrx9s6uW5H35LVtCGmVVs/5i8O8m4OSrrjVxfshbOvGfW0Kfp4KyVUzppPdEAuboVsx+23ZcV4cCsEaYvoC2Fqgn5t9wrLqA2WCPUQztdH0NEkP6bGDlTTlsOImBnrbgIqoHrV5ZHmcfGRpJ9LrcQBw2K/AuF7S9rHwyFdYwgDCC4jNPulQb7ZORdoFsj+AM/ymMZ/v3v/3F8Irni5ICKQRnXIgRcfTU3iyrG5IAJ8DXETS5aqLwoY7uMCRBwf+yGV13gLcvVCLQs9qAaaRyCQEF3ut8sB4gDZKTTmQwPEXtyuJBIQmgc4iYzflQYPJkOShjg0naokhqmkpAntEAKowtC9ikWZG7FyxH1tlxFYqIb23pfe+sd4weqEnoJDzn/ADEDB5hLs78Dz6QDOEo38fATjRE5zBvFYA7lPa7MBYaH9fg8A88dG1kHkB0cN34rRLR3A334DE0T5r4RuaerHDC71NTZE9CcPaQn1nG/HGqvBfRxtW/6mNGFmUFM37icxbpLublkuB88+sk6nMbQOVm/ZZQ1YvgAZZP+5IPUQ+B8Zp6bbn7qzPhenCNbnKcN0wun56dbwJpGpzeGKG4IZa0N9j1gm9v6BvY/VVw36W748wcEFmTqGEEYs/Q3ZOlvZOAlchgyzz0U3Rr79QX6jnThITt1GEcZv2US3VCdD7sqzsSgCdA1xAAwIRasIx2LdqipBN3hvErDVT3SkluaqxFsafxOZdI4BIGC7lLfSf0wmehkeFenI/6FRIKhcfcLn/FuIQoDPCE
```
Comment (MartinKolarik) [inline src/public/img/about/about-gradient.svg:7]: There's no point in making this svg when it's just a base64 encoded png inside. Use the png directly and delete this.

Comment (xbpcb) [reply src/public/img/about/about-gradient.svg]: Replaced

File: src/views/components/landing-top.html [hunk 2]
```diff
@@ -4,303 +4,174 @@
...
+			</div>
+			<div>
+				<h3>over {{_.formatMiB2TiB(networkStats.megabytes.total)}} TiB</h3>
+				<span>last month</span>
+			</div>
+		</section>
 
 
 		<section class="docs">
-			<div class="docs-tabs">
```
Comment (MartinKolarik) [inline src/views/components/landing-top.html:28]: Why are we changing from `docs-tabs` to `docs_tabs` in some places? I believe we used hyphens everywhere until now.

File: src/views/components/landing-top.html [hunk 3]
```diff
@@ -337,6 +208,228 @@ <h3>over {{_.formatMiB2TiB(networkStats.megabytes.total)}} TiB</h3>
 					{ slug: 'gh', label: 'GitHub', image: 'github.svg' },
 					{ slug: 'wp', label: 'WordPress', image: 'wordpress.svg' },
 				],
+				slidesData: {
```
Comment (MartinKolarik) [inline src/views/components/landing-top.html:211]: Why this change? It's probably harder to edit than before.

**Review by MartinKolarik — CHANGES_REQUESTED** [review]
This PR introduces new bugs and reverts some of the previous fixes. It fixes things too, but right now, I can't tell if it's overall better or worse.

I believe this is partly because we already performed a few rounds of QA on the base PR and made small modifications from the design in Figma, e.g., to improve how things work on smaller screens. All these changes were discussed in #400 and #438 (note that there are many comments and some have partially hidden content under "Details"). Examples of these issues:
 - "Top 10 npm packages last 30 days" is more narrow, which means its content fits poorly on smaller screens (maybe it could stay this size on xl, but on smaller, it needs to take more % of the width
 - Fonts for table numbers were different on purpose because the default font doesn't work well for numbers

I know it's going to be rather complicated, but I'll have to ask you to go through your changes and check that if you change something that doesn't match Figma, it also wasn't mentioned in my previous comments in the linked issues.

The second type of issues is simply bugs that can be resolved with more testing (on all screen sizes), e.g.:
 - https://dl.dropboxusercontent.com/s/1ldbd3k6n9wej2e/2022-07-22_14-36-54.png
 - https://dl.dropboxusercontent.com/s/xug1pharzr5ft7v/2022-07-22_14-38-11.png

File: src/views/components/stats-table.html [hunk 4]
```diff
@@ -3,19 +3,23 @@
...
-       <div class="table-chart-lg">
-        <div class="table-chart" style-width="{{this[valueKey] / data[desc ? 0 : data.length - 1][valueKey] * 100}}%"></div>
+      <div class="item">
+        {{#if rowIndexes}}
+          <span class="index">{{i+1}}</span>
+        {{/if}}
+
+        {{>column1}}
+
+        <span class="hits">{{formatValue(this[valueKey])}}</span>
```
[the remark this answers is not in the document]
Comment (xbpcb) [reply src/views/components/stats-table.html:17]: Ok, was not aware of these changes, did according to the design.
So now i am going to go through the list from #400  and #438 and implement these changes.

File: src/views/components/landing-top.html (hunk 2 above)
Comment (xbpcb) [reply src/views/components/landing-top.html:28]: Easier to read when nested elements separated by dash and the name parts with hyphens

File: src/views/components/landing-top.html (hunk 3 above)
Comment (xbpcb) [reply src/views/components/landing-top.html:211]: I believe that it is much easier to edit the data or extend the functionality if needed later when the data and the view are separated

**Comment by MartinKolarik** [comment]
Just to confirm, the gradient files were made smaller because parts of them are not actually visible?

**Comment by xbpcb** [comment]
> Just to confirm, the gradient files were made smaller because parts of them are not actually visible?

Hm, what do you mean by gradient files? I mean i've cut some of the images since they had too much empty space and additional empty space was not equilateral, but maybe i am thinking of the wrong files.

**Comment by MartinKolarik** [comment]
The first two here in the list of changed files.
